Sources/Diagnostics/FileHandler.swift

```diff
@@ -0,0 +1,256 @@
+                    .components(separatedBy: String(Self.lineBreak))
+                    .suffix(linesDetected - count + 1)
+                    .joined(separator: String(Self.lineBreak))
+
+                outputFile.write(linesToWrite.asData)
+            }
+        } while true
+
+        // Replace with temporary file
+        try self.replaceHandler(with: tempURL)
+    }
+
+    // MARK: -
+
+    private static let fileManager: FileManager = .default
+
+    private static let lineBreak: Character = "\n"
+    private static let lineBreakData = String(FileHandler.lineBreak).asData
+    private static let bufferSize = 4096
+
+}
+
+// MARK: - Errors
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+extension FileHandler {
+
+    enum Error: Swift.Error {
+
+        case failedCreatingFile(URL)
+        case failedCreatingDirectory(URL)
+        case failedCreatingHandle(Swift.Error)
+        case failedSeeking(Swift.Error)
+        case failedEmptyingFile(Swift.Error)
+        case failedMovingNewFile(from: URL, toURL: URL, Swift.Error)
+
+    }
+
+}
+
+// MARK: - Private
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+private extension FileHandler {
+
+    func moveToBeginningOfFile() throws {
+        do {
+            try self.fileHandle.seek(toOffset: 0)
+        } catch {
+            throw Error.failedSeeking(error)
+        }
+    }
+
+    static func createFileIfNecessary(_ url: URL) throws {
+        guard !Self.fileManager.fileExists(atPath: url.path) else { return }
+
+        let directoryURL = url.deletingLastPathComponent()
+        if !Self.fileManager.fileExists(atPath: directoryURL.path) {
+            do {
+                Logger.verbose(Message.creating_directory(directoryURL))
+
+                try Self.fileManager.createDirectory(at: directoryURL,
+                                                     withIntermediateDirectories: true,
+                                                     attributes: nil)
+            } catch {
+                throw Error.failedCreatingDirectory(directoryURL)
+            }
+        }
+
+        Logger.verbose(Message.creating_file(url))
+
+        if !Self.fileManager.createFile(atPath: url.path, contents: nil, attributes: nil) {
+            throw Error.failedCreatingFile(url)
+        }
+    }
+
+    static func createTemporaryFile() throws -> URL {
+        let result = Self.fileManager.temporaryDirectory
+            .appendingPathComponent("com.revenenuecat")
+            .appendingPathComponent(UUID().uuidString)
+
+        try Self.createFileIfNecessary(result)
+        return result
+    }
+
+    func replaceHandler(with otherURL: URL) throws {
+        do {
+            try Self.fileManager.removeItem(at: self.url)
+            try Self.fileManager.moveItem(at: otherURL, to: self.url)
+        } catch {
+            throw Error.failedMovingNewFile(from: otherURL, toURL: self.url, error)
+        }
+
+        self.fileHandle.closeAndLogErrors()
```

Comment:
So closing the fileHandle AFTER we've removed the original item won't be a problem? I'd imagine we would need to close it before deleting the file but I might be wrong.

Reply:
This is basically just a file descriptor inside. I did this after moving it because we don't want to leave an inconsistent state if any of the move operations fail.